### PR TITLE
State variable pointers should never change.

### PR DESCRIPTION
**Pull request by insperatum**
see: https://github.com/torch/nn/blob/master/doc/module.md#state-variables

### CODE REVIEW

**Comment by soumith** [comment]
Thanks a lot. There's a few more modules where this bad habit has propagated.

**Comment by koraykv** [comment]
What if input to nn.Identity is not a tensor?

**Comment by soumith** [comment]
oh, I didn't know that we are using it in such contexts. Let me revert the PR, sorry about that.

**Comment by insperatum** [comment]
Aah whoops, sorry guys! My first commit and I accidentally run the test suite on the unmodified code :/
Will be more careful next time.

What do you think is the correct behaviour then, in modules which accept both tables and tensors? Hold onto one pointer for each type, or just keep it as it is and forget about preserving the pointer?

**Comment by koraykv** [comment]
I would suggest checking if the input is a tensor and then applying your version. For anything else, I think it is too involved. It might be arbitrary data passed around with hierarchies of tables...

File: Identity.lua [hunk 1]
```diff
@@ -1,12 +1,12 @@
 local Identity, _ = torch.class('nn.Identity', 'nn.Module')
 
 function Identity:updateOutput(input)
-   self.output = input
+   self.output:set(input:storage(), input:storageOffset(), input:size(), input:stride())
```
Comment (dominikgrewe) [inline Identity.lua]: `self.output:set(input)` has the same effect.

**Pull request merged**